app/client/components/DetailView.js

```diff
@@ -76,17 +76,33 @@ function DetailView(gristDoc, viewSectionModel) {
   //--------------------------------------------------
   // Set up DOM event handling.
 
+  this.lastElemSelected = null;
+
   // Clicking on a detail field selects that field.
   this.onEvent(this.viewPane, 'mousedown', '.g_record_detail_el', function(elem, event) {
     this.viewSection.hasFocus(true);
     var rowModel = this.recordLayout.getContainingRow(elem, this.viewPane);
     var field = this.recordLayout.getContainingField(elem, this.viewPane);
     commands.allCommands.setCursor.run(rowModel, field);
+    this.lastElemSelected = elem;
   });
 
   // Double-clicking on a field also starts editing the field.
   this.onEvent(this.viewPane, 'dblclick', '.g_record_detail_el', function(elem, event) {
     this.activateEditorAtCursor();
+    this.lastElemSelected = elem;
+  });
+
+  // Clicking on a field also starts editing the field but only on certain cases and if the field are preselected
+  const simpleClickUnacceptedSelectors = [".formula_field", ".widget_checkbox", ".widget_switch"]
+  this.onEvent(this.viewPane, 'click', '.g_record_detail_el', function(elem, event) {
+    if (
+      elem === this.lastFieldSelected
```

Comment:
Can you disable this behavior on a mobile screen (using `isNarrowScreen` helper)? It won't be convenient for mobile users to accidentally trigger the virtual keyboard.

Reply:
I can add a check with `isNarrowScreen` but the double click is seems not working in mobile screen and personally it seems to me an unusual behavior no ?

Reply:
@berhalak

Reply:
Sorry for the delay. I'm not exactly sure what you mean by "double click is not working on mobile". It works fine for me, and I'm using Grist primarily on mobile.

Reply:
I try the responsive on my browser and double click on fields and nothing happened, maybe because of my browser